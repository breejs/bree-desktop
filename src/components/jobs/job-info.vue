<script setup>
import { computed } from 'vue';

const props = defineProps({
  name: {
    type: String,
    required: true
  },
  lastRun: Date
});

const lastRun = computed(() => {
  return props.lastRun?.toLocaleString() || 'never';
});
</script>

<template lang="pug">
.row
  h4= '{{ props.name }}'
.row
  p.text-muted= 'Last run: {{ lastRun }}'
</template>
