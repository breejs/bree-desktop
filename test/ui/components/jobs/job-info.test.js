import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/job-info.stories';

storyTests(stories);
