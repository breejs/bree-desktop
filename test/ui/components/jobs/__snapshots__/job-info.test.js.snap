// Vitest Snapshot v1

exports[`story snapshots > WithLastRun 1`] = `
"<div class=\\"row\\">
  <h4>Job Name</h4>
</div>
<div class=\\"row\\">
  <p class=\\"text-muted\\">Last run: 4/4/2022, 12:00:00 PM</p>
</div>"
`;

exports[`story snapshots > WithNoLastRun 1`] = `
"<div class=\\"row\\">
  <h4>Job Name</h4>
</div>
<div class=\\"row\\">
  <p class=\\"text-muted\\">Last run: never</p>
</div>"
`;
